scripts/apps/desks/controllers/DeskConfigController.ts

```diff
@@ -1,24 +1,31 @@
 import _ from 'lodash';
 import {gettext} from 'core/utils';
 
-DeskConfigController.$inject = ['$scope', '$controller', 'notify', 'desks', 'WizardHandler', 'modal'];
-export function DeskConfigController($scope, $controller, notify, desks, WizardHandler, modal) {
+DeskConfigController.$inject = ['$scope', '$controller', 'notify', 'desks', 'modal', '$rootScope'];
+export function DeskConfigController($scope, $controller, notify, desks, modal, $rootScope) {
     // expecting $scope.desks to be defined
 
+    const stepsToWatch = ['general', 'people'];
+
     $scope.modalActive = false;
     $scope.numberOfUsers = 3;
     $scope.step = {
         current: null,
     };
     $scope.desk = {
         edit: null,
+        orig: null,
     };
 
     $scope.openDesk = function(step, desk) {
-        $scope.modalActive = true;
-        $scope.step.current = step;
-        $scope.desk.edit = desk || {};
-        $scope.desk.edit.desk_metadata = $scope.desk.edit.desk_metadata || {};
+        desks.fetchDesks().then((_desks) => {
```

Comment:
Why do you need to fetch it again? If you really need it, it's better to fetch only the one you need, not all of them. You can do this easily by using `dataApi.findOne<IDesk>(desk._id)`.

You shouldn't try to set a default value like `const updatedDesk = ... || {};`. If I understand correctly, the desk must be there. If it's not for some reason, log the error `logger.error` and exit. You may display a message to the user, but do not continue with the program when you know you have corrupted data.

Reply:
Just to make sure there is no `_etag` mismatch error when we edit the desk again without refreshing and will change it to find only one desk.